### PR TITLE
Move module signature loading to the oak app builder

### PR DESCRIPTION
**Pull request by conradgrobler**
This change moves the logic for loading module signature to the oak app builder. The singatures are now bundled into the application configuration file as part of the app build process.

This is a step towards supporting #1780

This change also adds support for downloading signatures from a URL (see #1379)

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [x] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: sdk/rust/oak_app_build/src/main.rs [hunk 1]
```diff
@@ -233,6 +263,28 @@ async fn main() -> anyhow::Result<()> {
         modules.insert(name.to_string(), loaded_module);
     }
 
+    // Load signatures.
+    let mut signatures = Vec::new();
+    for location in manifest.signature_manifests.iter() {
```
Comment (ipetr0v) [inline sdk/rust/oak_app_build/src/main.rs:268]: Should the variable be called `manifest_file` instead? Since it may be confusing with `signature_location`

**Review by ipetr0v — APPROVED** [review]

File: sdk/rust/oak_app_build/src/main.rs (hunk 1 above)
Comment (conradgrobler) [reply sdk/rust/oak_app_build/src/main.rs]: Renamed the variable to `signature_manifest_location` to make the intention clearer.

File: oak_abi/proto/application.proto [hunk 2]
```diff
@@ -48,6 +48,16 @@ message ApplicationConfiguration {
   // Usually a NodeConfiguration entry that holds a WebAssemblyConfiguration
   // object.
   NodeConfiguration initial_node_configuration = 2;
+
+  // The bundled signatures for validating Wasm module hashes.
+  repeated ModuleSignature module_signatures = 3;
+}
+
+// Bundled signature of the SHA-256 hash of a Wasm module.
+message ModuleSignature {
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply oak_abi/proto/application.proto:57]: The closest I could find was `SignedChallenge` that is used for auth. It does not include the hash field. Do you think we should make a more generic "signed data" proto and use it for both?

[the remark this answers is not in the document]
Comment (conradgrobler) [reply oak_abi/proto/application.proto:58]: I created #1830 to track this.

**Review by tiziano88 — APPROVED** [review]

**Pull request merged**